.github/workflows/ci.yml

```diff
@@ -26,16 +26,19 @@ jobs:
         git config --global user.email "x@y.z"
     - name: Checkout Commit
       uses: actions/checkout@v1
+    - name: checkout main
+      run: git branch -f main origin/main
```

Comment:
Why do you need to checkout main? What if you are on a branch? The tests should run on the branch not on main

Reply:
https://github.com/pnpm/pnpm/pull/4047/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR49 here will judge the branch?